pkg/build/graph/analysis/bc.go

```diff
@@ -68,6 +76,60 @@ bc:
 	return markers
 }
 
+// FindMissingInputImageStreams checks all build configs and confirms that their From element exists
+//
+// Precedence of failures:
+// 1. A build config's input points to an image stream that does not exist
+// 2. A build config's input uses an image stream tag reference in an existing image stream, but no images within the image stream have that tag assigned
+// 3. A build config's input uses an image stream image reference in an exisiting image stream, but no images within the image stream have the supplied image hexadecimal ID
+func FindMissingInputImageStreams(g osgraph.Graph) []osgraph.Marker {
+	markers := []osgraph.Marker{}
+
+	for _, bcNode := range g.NodesByKind(buildgraph.BuildConfigNodeKind) {
+		for _, istNode := range g.PredecessorNodesByEdgeKind(bcNode, buildedges.BuildInputImageEdgeKind) {
+			for _, uncastImageStreamNode := range g.SuccessorNodesByEdgeKind(istNode, imageedges.ReferencedImageStreamGraphEdgeKind) {
+				imageStreamNode := uncastImageStreamNode.(*imagegraph.ImageStreamNode)
+
+				// note, BuildConfig.Spec.BuildSpec.Strategy.[Docker|Source|Custom]Stragegy.From Input of ImageStream type will get converted to ImageStreamTag on the vX to api conversion
+				if !imageStreamNode.Found() {
+					break
+				}
+
+				if tagNode, ok := istNode.(*imagegraph.ImageStreamTagNode); ok {
+					imageStream := imageStreamNode.Object().(*imageapi.ImageStream)
+					if _, ok := imageStream.Status.Tags[tagNode.ImageTag()]; !ok {
+						addInputMarker(&markers, bcNode, istNode, imageStreamNode, MissingImageStreamTagWarning,
+							fmt.Sprintf("%s builds from %s, but the image stream tag does not exist.", bcNode.(*buildgraph.BuildConfigNode).ResourceString(), istNode.(*imagegraph.ImageStreamTagNode).ResourceString()),
+							osgraph.Suggestion(fmt.Sprintf("examine analysis of build config outputs from this command and see if they build %s", istNode.(*imagegraph.ImageStreamTagNode).ResourceString())))
+					}
+				}
+			}
+
+			for _, uncastImageStreamNode := range g.SuccessorNodesByEdgeKind(istNode, imageedges.ReferencedImageStreamImageGraphEdgeKind) {
+				imageStreamNode := uncastImageStreamNode.(*imagegraph.ImageStreamNode)
+
+				// note, BuildConfig.Spec.BuildSpec.Strategy.[Docker|Source|Custom]Stragegy.From Input of ImageStream type will get converted to ImageStreamTag on the vX to api conversion
+				if !imageStreamNode.Found() {
```

Comment:
That comment was a copy/paste error ... it was only relevant to ImageStreamTag ... also fixed the tense, elaborated slightly, to better convey that the conversion has in fact already happened as you say